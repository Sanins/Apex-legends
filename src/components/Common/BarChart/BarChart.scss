@import './../../Common/colours.scss';

.bar-chart {
    width: 100%;
    text-align: center;
    &__svg {
        width:100px;
        height:20px;
        padding-top: 7px;
        
        &__background {
            fill: $grey;
            height: 10px;
        }
        &__live-data { 
            fill: $secondary; 
            height: 10px;
        }
    }
}