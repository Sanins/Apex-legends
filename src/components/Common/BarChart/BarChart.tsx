import React, { useState, useEffect } from 'react';
import './BarChart.scss';

interface BarChartProps {
    dataTitle: any;
    dataValue: any;
    dataName?: any;
}

export default function BarChart(props: BarChartProps) {
    const [highestValue, setHighestValue] = React.useState(0);

    useEffect(() => {
        switch (props.dataTitle) {
            case 'Damage per second':
                return setHighestValue(255);
            case 'Headshot damage':
                return setHighestValue(110);
            case 'Legshot damage':
                return setHighestValue(44);
            case 'Damage per mag':
                return setHighestValue(660);  
            case 'Average dps/sec':
                return setHighestValue(255);  
            case 'Headshot dps':
                return setHighestValue(400);  
            case 'Legshot dps':
                return setHighestValue(200);       
        };
    });

    const trueValue = ((props.dataValue / highestValue) * 100).toFixed(2); 

    return (
        <div className='bar-chart'>
            <svg className='bar-chart__svg'>
                <g>
                    <rect className='bar-chart__svg__background' width='100%' ></rect>
                    <rect className='bar-chart__svg__live-data' width={trueValue + '%'}></rect>
                </g>
            </svg>
        </div>
    );
}
