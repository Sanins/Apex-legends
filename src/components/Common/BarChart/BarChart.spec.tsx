/* eslint-disable no-undef */
import React from "react";
import { mount, shallow } from "enzyme";
import BarChart from "./BarChart";
import toJson from "enzyme-to-json";

describe("BarChart", () => {
	it("should fetch data on mount", () => {
		const wrapper = shallow(<BarChart dataValue={50} dataTitle="Headshot dps" />);
		wrapper.update();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
