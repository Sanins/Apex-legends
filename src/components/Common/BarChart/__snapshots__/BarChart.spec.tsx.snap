// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart should fetch data on mount 1`] = `
<div
  className="bar-chart"
>
  <svg
    className="bar-chart__svg"
  >
    <g>
      <rect
        className="bar-chart__svg__background"
        width="100%"
      />
      <rect
        className="bar-chart__svg__live-data"
        width={0}
      />
    </g>
  </svg>
</div>
`;
